packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -210,6 +210,7 @@ export class FileNavigatorWidget extends AbstractNavigatorTreeWidget {
 
     protected updateSelectionContextKeys(): void {
         this.contextKeyService.explorerResourceIsFolder.set(DirNode.is(this.model.selectedNodes[0]));
+        this.contextKeyService.isFileSystemResource.set(FileStatNodeData.is(this.model.selectedNodes[0]));
```

Comment:
Suggestion: VSCode additionally tests whether the file service has a provider for the URI. See [here](https://github.com/microsoft/vscode/blob/1867e88862ca26343822231267f20b839daef65d/src/vs/workbench/common/contextkeys.ts#L254). We should likely do the same.

Reply:
@msujew 
Yes, this is true. If I'm not mistaken, in Theia this is indirectly done by FileSystemService that gives FileStats only to entires that has a FileSystemProvider for the given schema. So I skipped the additional provider check. However, I can add it here too if needed.

Reply:
Ok, that's fine. Do you mind adding a comment that explains this implicit behavior then?

Reply:
Comment is added